tests/ImageSharp.Tests/Formats/Jpg/JpegEncoderTests.cs

```diff
@@ -83,6 +83,12 @@ public void Encode_PreserveQuality(string imagePath, int quality)
         public void EncodeBaseline_WorksWithDifferentSizes<TPixel>(TestImageProvider<TPixel> provider, JpegSubsample subsample, int quality)
             where TPixel : unmanaged, IPixel<TPixel> => TestJpegEncoderCore(provider, subsample, quality);
 
+        [Theory]
+        [WithFile(TestImages.Png.BikeGrayscale, nameof(BitsPerPixel_Quality), PixelTypes.L8)]
+        [WithSolidFilledImages(nameof(BitsPerPixel_Quality), 1, 1, 100, 100, 100, 255, PixelTypes.L8)]
+        public void EncodeBaseline_GrayscaleWorksWithDifferentSizes<TPixel>(TestImageProvider<TPixel> provider, JpegSubsample subsample, int quality)
```

Comment:
We also need test cases for:
- `PixelTypes.Rgba32` (general case with explicit setting of  `colorType:JpegColorType.Luminance)`, the actual image can be grayscale, so the validation can be done easily
- A case with `PixelTypes.L8`, but  `colorType:null` to test the automatic detection of `ColorType` when not specified

Reply:
All these test cases added. Renamed the new testcase, as now it is fully focussed on Grayscale.